### PR TITLE
Closes #3510-Series.to_pandas to handle categoricals

### LINKED ISSUE #3510: Series.to_pandas to handle categoricals

### PR DESCRIPTION
**Pull request by ajpotts**
Closes #3510-Series.to_pandas to handle categoricals

### CODE REVIEW

File: PROTO_tests/tests/series_test.py [hunk 1]
```diff
@@ -41,6 +43,48 @@ def test_series_creation(self, dtype):
...
+        ak_cat = Series(pd_cat)
+        ak_num = Series(pd_num)
+
+        expected_str = Series(str_vals, index=idx)
+        expected_cat = Series(cat_vals, index=idx)
+        expected_num = Series(num_vals, index=idx)
+
+        ak_assert_series_equal(ak_str, expected_str)
+        ak_assert_series_equal(ak_cat, expected_cat)
+        ak_assert_series_equal(ak_num, expected_num)
```
Comment (stress-tess) [inline PROTO_tests/tests/series_test.py:67]: this doesn't matter at all, but we could reduce some code duplication:
```python
  vals = [ak.random_strings_uniform(9, 10, size), ak.Categorical(str_vals), ak.arange(size) * -2]
  for val in vals:
    idx = ak.arange(size) * -1
    pd_ser = pd.Series(val.to_ndarray(), idx.to_ndarray())
    ak_ser = Series(pd_ser)
    expected = Series(val, index=idx)
    ak_assert_series_equal(ak_ser, expected)
```

File: PROTO_tests/tests/series_test.py [hunk 2]
```diff
@@ -41,6 +43,48 @@ def test_series_creation(self, dtype):
...
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_to_pandas(self, size):
```
Comment (stress-tess) [inline PROTO_tests/tests/series_test.py:70]: same here

**Review by stress-tess — APPROVED** [review]
looks good! I left some suggestions on the tests that are aesthetic

**Review by jaketrookman — APPROVED** [review]
Besides the suggestion to make the test a little more succinct, looks good

File: PROTO_tests/tests/series_test.py (hunk 2 above)
Comment (ajpotts) [reply PROTO_tests/tests/series_test.py:62]: Made the change.  Thanks.

File: PROTO_tests/tests/series_test.py (hunk 1 above)
Comment (ajpotts) [reply PROTO_tests/tests/series_test.py]: I had to make some adjustments so it would run, but I did make the change.  Thanks.

**Review by stress-tess — APPROVED** [review]

**Pull request merged**